dot/state/storage_notify.go

```diff
@@ -33,55 +34,105 @@ type SubscriptionResult struct {
 	Changes []KeyValue
 }
 
-//StorageSubscription holds data for Subscription to Storage
-type StorageSubscription struct {
-	Filter   map[string]bool
-	Listener chan<- *SubscriptionResult
+// Observer interface defines functions needed for observers, Observer Design Pattern
+type Observer interface {
+	Update(result *SubscriptionResult)
+	GetID() int
```

Comment:
bit of a nitpick but maybe change int to uint?

Reply:
updated.